src/Raven.Server/Web/System/AdminDatabasesHandler.cs

```diff
@@ -828,6 +879,185 @@ await using (var writer = new AsyncBlittableJsonTextWriter(context, ResponseBody
+            Dictionary<string, string> clusterNodes, // (nodeTag, url)
+            HashSet<string> databaseNodes, // nodeTags
+            CancellationToken token)
+        {
+            try
+            {
+
+                foreach (var (tag, nodeUrl) in clusterNodes)
+                {
+                    if (tag == ServerStore.NodeTag)
+                    {
+                        await ServerStore.WaitForCommitIndexChange(RachisConsensus.CommitIndexModification.GreaterOrEqual, index);
+
+                        if (databaseNodes.Contains(tag))
+                        {
+                            try
+                            {
+                                var database = await ServerStore.DatabasesLandlord.TryGetOrCreateResourceStore(databaseName);
+                                if (database != null) // database isn't deleted already from current node 
+                                {
+                                    await database.RachisLogIndexNotifications.WaitForIndexNotification(index, token);
+                                }
+                            }
+                            catch (Exception ex) when (ex is DatabaseDoesNotExistException or
+                                                           DatabaseNotRelevantException or
+                                                           DatabaseDisabledException)
+                            {
+                            }
+                        }
+
+                    }
+                    else
+                    {
+                        try
+                        {
+                            using (var requestExecutor =
+                                   ClusterRequestExecutor.CreateForSingleNode(nodeUrl, Server.Certificate.Certificate, DocumentConventions.DefaultForServer))
+                            using (requestExecutor.ContextPool.AllocateOperationContext(out var context))
+                            {
+                                try
+                                {
+                                    if (databaseNodes.Contains(tag)) // if node is part of db topology - wait for commit and database notification
+                                    {
+                                        var cmd = new WaitForDatabaseIndexNotificationCommand(index, databaseName);
+                                        await requestExecutor.ExecuteAsync(cmd, context, token: token);
+                                    }
+                                    else // else - wait for commit only
+                                    {
+                                        var cmd = new WaitForCommitCommand(index);
+                                        await requestExecutor.ExecuteAsync(cmd, context, token: token);
+                                    }
+                                }
+                                catch (Exception ex) when (ex is DatabaseDoesNotExistException or
```

Comment:
I don't expect any exceptions here, they should be handled on the destination EP

Reply:
Done